pkg/util/clusterauthorizer/authorizer.go

```diff
@@ -71,6 +73,7 @@ func (a *azRefreshableAuthorizer) NewRefreshableAuthorizerToken(ctx context.Cont
 	c := &azureclaim.AzureClaim{}
 	_, _, err = p.ParseUnverified(token.OAuthToken(), c)
 	if err != nil {
+		err = api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidServicePrincipalToken, "properties.servicePrincipalProfile", err.Error())
```

Comment:
How did you confirm that all `token contains` errors come from `p.ParseUnverified(token.OAuthToken(), c)`? Mostly just curious.

Reply:
Searching for that string in our repo (excluding vendor/*) results in one result from a test. If you search for this string '"The provided service principal generated an invalid token."' you find one result in `aad.go`. The real failure to parse token error comes from [parser.go](https://github.com/Azure/ARO-RP/blob/master/vendor/github.com/form3tech-oss/jwt-go/parser.go#L99).

That makes me believe it's the source of all those errors.